contrib/NeonVM/stack.c

```diff
@@ -64,11 +66,27 @@ void pop(struct tagTStack *stack)
     cell_freeCell(p);
 }
 
-Cell *top(struct tagTStack *stack)
+Cell *top(TStack *stack)
 {
     if (isEmpty(stack)) {
         fatal_error("Stack underflow error.");
     }
 
     return stack->data[stack->top];
 }
+
+Cell *peek(TStack *stack, int element)
+{
+    if (isEmpty(stack)) {

```

Comment:
Wouldn't this situation be handled correctly by the next test? If the stack is empty, then stack->top is zero, and (stack->top - element) will be less than zero.